tox.ini

```diff
@@ -166,7 +184,11 @@ allowlist_externals =
     mkdir
     cp
 deps =
-    -e .[testing]
+    # TODO: If the code in testing changes, it isn't reinstalled.
+    # -e doesn't install it with the right name. We should probably
+    # build a wheel and then use that across the environments,
+    # similar to what was done in the ops-scenario repository.
+    ./testing
```

Comment:
Is this set of deps intentionally missing `-e .`?

Reply:
Yes, see the comment above it :wink:.

Reply:
Oh wait, that wasn't "I didn't read the comment", that was "-e .", sorry!

Reply:
Yeah I just noticed that this one was different from the others where you made similar changes. This one is missing the `-e .` line that the others replaced `-e .[testing]` with.

Reply:
And the answer is yes, because it's only testing the `testing/README.md` file. Although thinking about it more, I suppose it ought to have the latest ops code as well, so I'll add it in.

I think we'll probably end up shrinking `testing/README.md` down with the examples there moving into how-tos (some has already) and the API reference docs (also) and so on, and then can remove this tox environment. It was super handy while I was doing the 7.0 work to avoid forgetting to update the README or doing it incorrectly, even though it's limited in what it checks.

Reply:
Makes sense! Code snippets in readmes are nice, and making sure they run correctly is nice too.